homeassistant/components/hdmi_cec/__init__.py

```diff
@@ -374,10 +374,27 @@ class CecEntity(Entity):
     def __init__(self, device, logical) -> None:
         """Initialize the device."""
         self._device = device
-        self._icon = None
         self._state: str | None = None
         self._logical_address = logical
         self.entity_id = "%s.%d" % (DOMAIN, self._logical_address)
+        self._attr_should_poll = False
+        self._attr_name = (
+            f"{self.vendor_name} {self._device.osd_name}"
+            if (
+                self._device.osd_name is not None
+                and self.vendor_name is not None
+                and self.vendor_name != "Unknown"
+            )
+            else "%s %d" % (self._device.type_name, self._logical_address)
+            if self._device.osd_name is None
+            else "%s %d (%s)"
+            % (self._device.type_name, self._logical_address, self._device.osd_name)
+        )
```

Comment:
putting such extensive logic in the constructor, doesn't make this more readable IMHO.

Reply:
I am not sure it is more readable, but now the logic is in its own method.